### ISSUE TITLE
error : "Error database insert failed" mid Dec onwards on multiple machines same error

### ISSUE THREAD

**Issue by vistisoara** [fwenzel/copy-shorturl#150]
hi, Copy ShortURL firefox add-in worked fine all during december, then it started only working intermitently for 1-2 days and then it stopped working altogether; now it gives me the "Error databse insert failed"; I have tried clearing the cache, has not made any difference.

**Comment by ptarsus07** [comment]
Same circumstance here.  Copy ShortURL no longer working in any Firefox mode (i.e., including "Private").  This appears to have started when Firefox was auto-updated to v121.0.1 (64-bit); however, that might be a coincidence.

Running v5.5.0 (updated 01/10/24)

Options selected:
- automatic updates = default
- run in private windows = allow
- run on restricted websites = don't allow

Permissions granted:
- access data for all websites
- input data to the clipboard
- display notifications

Preferences selected:
- no URL shortening service selected
- shorten canonical URL when longer than 30 characters
- use special case URL formats (when available)
- remove ?utm_* parameters
- show notification pop-up

Preferences not selected:
- keep #hash in final URL
- copy page title

Other add-ons operating in the environment are:
- DuckDuckGo Privacy Essentials
- Easy Youtube Video Downloader Express
- uBlock Origin

Other environment information:
- Apple MacBook Air (M1, 2020 w/ 16Gb memory)
- macOS Sonoma (v14.2.1 23C71)

Please note / notify when repaired...

Thanks!

**Comment by fwenzel** [comment]
Oof! I wonder if there was a breaking Firefox change that caused this. The extension itself did not change. I'll investigate and roll an update as needed!

**Comment by fwenzel** [comment]
@vistisoara @ptarsus07 what shortening service do you use? I use bitly and it appears to work just fine.

**Comment by ss249tp** [comment]
> @vistisoara @ptarsus07 what shortening service do you use? I use bitly and it appears to work just fine.

Hi @fwenzel I suspect there is some problem with the default shortener is.gd. For the past two months or so it often displays the same error message ("Error, database insert failed") even when I use it directly. This is unfortunate because I cannot find another shortening service that is as clean and limit-free as is.gd.

**Comment by fwenzel** [comment]
@ss249tp I agree - it does not seem to be a problem with the extension, more with the service used. I should make the extension make that clear, I admit that it is confusing.

For now my recommendation is to use a different shortening service (I use bitly with an account and API key but it is not a paid plan). The other shorteners seem to work fine also, I guess it depends on what one is looking for.
